File: frontend/src/app/models/order.dto.ts
export interface OrderDto {
  id: number;
  userId: string;
  items: OrderItemDto[];
  total: number;
  status: OrderStatus;
  shippingAddress: string;
  shippingMethod: string;
  trackingNumber: string;
  paymentId: string;
  createdAt: string;
  updatedAt: string;
}

export interface OrderItemDto {
  productId: number;
  quantity: number;
  unitPrice: number;
}

export enum OrderStatus {
  PENDING = 'PENDING',
  PAID = 'PAID',
  SHIPPED = 'SHIPPED',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED'
}

File: backend/ecovida-order-service/src/main/java/org/ecovida/order/service/dto/OrderDto.java
package org.ecovida.order.service.dto;

import lombok.*;
import org.ecovida.order.service.model.OrderStatus;

import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderDto {
    private Long id;
    private String userId;
    private List<OrderItemDto> items;
    private Double total;
    private OrderStatus status;
    private String shippingAddress;
    private String shippingMethod;
    private String trackingNumber;
    private String paymentId;
    private Date createdAt;
    private Date updatedAt;
}
